homeassistant/components/unifiled/light.py

```diff
@@ -0,0 +1,113 @@
+"""Support for Unifi Led lights."""
+import logging
+
+import voluptuous as vol
+
+# Import the device class from the component that you want to support
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    PLATFORM_SCHEMA,
+    SUPPORT_BRIGHTNESS,
+    Light,
+)
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+# Validation of the user's configuration
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_USERNAME): cv.string,
+        vol.Optional(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_PORT, default="20443"): cv.string,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Unifi LED platform."""
+
+    from unifiled import unifiled
+
+    # Assign configuration variables.
+    # The configuration check takes care they are present.
+    _ip = config[CONF_HOST]
+    _port = config[CONF_PORT]
+    _username = config[CONF_USERNAME]
+    _password = config.get(CONF_PASSWORD)
+
+    api = unifiled(_ip, _port, username=_username, password=_password)
+
+    # Verify that passed in configuration works
+    if not api.getloginstate():
+        _LOGGER.error("Could not connect to unifiled controller")
+        return
+
+    # Add devices
+    add_entities(
+        UnifiLedLight(light, _ip, _port, _username, _password)
+        for light in api.getlights()
+    )
+
+
+class UnifiLedLight(Light):
+    """Representation of an unifiled Light."""
+
+    def __init__(self, light, ip, port, username, password):
+        """Init Unifi LED Light."""
+
+        from unifiled import unifiled
+
+        self._api = unifiled(ip, port, username=username, password=password)
```

Comment:
Maybe consider passing in the instance you created up above?

Reply:
Done, moving the import made this possible.